hvpy/utils.py

```diff
@@ -15,11 +15,13 @@ def decorator(func):
     return decorator
 
 
-def convert_date_to_isoformat(v: datetime) -> str:
+def convert_date_to_isoformat(v: datetime) -> Union[str, None]:
     """
     Converts the date from a datetime object to a string in the ISO format.
     """
-    return v.isoformat() + "Z"
+    if v:
+        return v.isoformat() + "Z"
+    return None
```

Comment:
What happened here?

Reply:
The parameter `reqObservationDate` is Optional DateTime, so when we pass `None` - it throws an error saying, None has no attribute called `isoformat`.

Reply:
Makes sense.